dumper/lib-ext-hive-metastore/src/main/java/com/google/edwmigration/dumper/ext/hive/metastore/HiveMetastoreThriftClient_v2_3_6.java

```diff
@@ -524,6 +524,21 @@ public Integer getCreateTime() {
     return out;
   }
 
+  @Override
+  public ImmutableList<String> getFunctionsAsJsonl() throws Exception {
+    TSerializer serializer = createJsonSerializer();
+    return client.get_all_functions().getFunctions().stream()
+        .map(
+            function -> {
+              try {
+                return serializer.toString(function);
+              } catch (TException e) {
+                throw new IllegalStateException(e);
+              }
+            })
+        .collect(toImmutableList());
```

Comment:
I would suggest moving the serialization part to the new static function that accepts `List<TBase>` as a parameter and returns `ImutableList<String>`. This can help to reduce code duplication.

Reply:
Agree; this should be in the file-writer (the task), not in the data-retriever/model.

Reply:
Done.